src/Files.App/Utils/Cloud/CloudDrivesDetector.cs

```diff
@@ -345,5 +329,33 @@ private static Task<IEnumerable<ICloudProvider>> DetectAutodeskDrive()
 
 			return Task.FromResult<IEnumerable<ICloudProvider>>(results);
 		}
+
+		private static string GetDriveType(string driveIdentifier, RegistryKey? namespaceSubKey, RegistryKey? syncRootManagerKey)
+		{
+			var appNameFromNamespace = (string?)namespaceSubKey?.GetValue("ApplicationName");
+
+			// Drive specific
+			if (driveIdentifier.StartsWith("iCloudDrive"))
+				return "iCloudDrive";
+			if (driveIdentifier.StartsWith("iCloudPhotos"))
+				return "iCloudPhotos";
+			if (driveIdentifier.StartsWith("ownCloud"))
+				return "ownCloud";
+			if (driveIdentifier.StartsWith("ProtonDrive"))
+				return "ProtonDrive";
+
+			// Nextcloud specific
+			if (!string.IsNullOrEmpty(appNameFromNamespace) && appNameFromNamespace == "Nextcloud")
+				return appNameFromNamespace;
```

Comment:
The order of Nextcloud specific and Drive specific are swapped, is it no problem? If it is, L335 should be moved just before here.

Reply:
The order of the checks should not matter.
But now the registry only has to be used if it cannot be identified by the driveIdentifier.

I have moved L335 👍